test/functional/p2p_opportunistic_1p1c.py

```diff
@@ -375,6 +390,160 @@ def test_other_parent_in_mempool(self):
+        peer_normal.send_and_ping(msg_tx(low_fee_parent["tx"]))
+        assert_equal(node.getmempoolentry(orphan_tx.rehash())["ancestorcount"], 2)
+
+    @cleanup
+    def test_orphanage_dos_many(self):
+        self.log.info("Test that the node can still resolve orphans when peers are sending tons of orphans")
+        node = self.nodes[0]
+        node.setmocktime(int(time.time()))
+
+        peer_normal = node.add_p2p_connection(P2PInterface())
+
+        batch_size = 100
+        num_peers = 10
+        # Each of the num_peers peers creates a distinct set of orphans
+        many_orphans = [self.create_small_orphan() for _ in range(batch_size * num_peers)]
+
+        self.log.info(f"Send sets of {batch_size} orphans from {num_peers} DoSy peers (may take a while)")
+        for peernum in range(num_peers):
+            peer_doser_batch = node.add_p2p_connection(P2PInterface())
+            for tx in many_orphans[batch_size*peernum : batch_size*(peernum+1)]:
+                # Don't sync with ping or wait for responses, because it dramatically increases the
+                # runtime of this test.
+                peer_doser_batch.send_message(msg_tx(tx))
+
+        # Something was evicted
+        self.wait_until(lambda: len(node.getorphantxs()) < batch_size * num_peers)
+
+        self.log.info("Send an orphan from a non-DoSy peer. Its orphan should not be evicted.")
+        low_fee_parent = self.create_tx_below_mempoolminfee(self.wallet)
+        high_fee_child = self.wallet.create_self_transfer(
+            utxo_to_spend=low_fee_parent["new_utxo"],
+            fee_rate=20*FEERATE_1SAT_VB,
+            target_vsize=100000
+        )
+
+        # Announce
+        orphan_wtxid = high_fee_child["wtxid"]
+        orphan_tx = high_fee_child["tx"]
+        orphan_inv = CInv(t=MSG_WTX, h=int(orphan_wtxid, 16))
+
+        # Wait for getdata
+        peer_normal.send_and_ping(msg_inv([orphan_inv]))
+        node.bumpmocktime(NONPREF_PEER_TX_DELAY)
+        peer_normal.wait_for_getdata([int(orphan_wtxid, 16)])
+        peer_normal.send_and_ping(msg_tx(orphan_tx))
+
+        # Wait for parent request
+        parent_txid_int = int(low_fee_parent["txid"], 16)
+        node.bumpmocktime(NONPREF_PEER_TX_DELAY + TXID_RELAY_DELAY)
+        peer_normal.wait_for_getdata([parent_txid_int])
+
+        shared_orphans = [self.create_small_orphan() for _ in range(batch_size)]
+        self.log.info(f"Send the same {batch_size} orphans from {num_peers} DoSy peers (may take a while)")
+        for _ in range(num_peers):
+            peer_doser_shared = node.add_p2p_connection(P2PInterface())
+            for orphan in shared_orphans:
+                peer_doser_shared.send_message(msg_tx(orphan))
+
+        # Something was evicted; the orphanage does not contain all DoS orphans + the 1p1c child
+        self.wait_until(lambda: len(node.getorphantxs()) < batch_size * num_peers + len(shared_orphans) + 1)
```

Comment:
this is a buggy assertion because no evictions will be happening, you might just front-run validation and slip through on your local machine. see https://github.com/instagibbs/bitcoin/commit/fedea4a17b7fc4c442b0ad98b51b85ff93a55beb for what I think would be a valid assertion

Reply:
Added thanks!